examples/echo-nats/echo_test.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"github.com/nats-io/go-nats"
+	"github.com/stretchr/testify/assert"
+	"testing"
+	"time"
+)
+
+type NatsStub struct {
+	Sub   string
+	Resp  interface{}
+	Reply interface{}
+}
+
+func (n *NatsStub) Request(subject string, v interface{}, vPtr interface{}, timeout time.Duration) error {
+	n.Sub = subject
+	n.Resp = v
+	return nil
+}
+
+func (n *NatsStub) QueueSubscribe(subject, queue string, cb nats.Handler) (*nats.Subscription, error) {
+	return nil, nil
+}
+
+func TestHandleRequestSuccess(t *testing.T) {
```

Comment:
In general I'm ok with tests however this looks more like crap i need some test here you go. What about failure scenarios? What about scenarios with something else where trigger is eg `nats echos ...` `nats echo ____my message` ___ indicates multiple spaces.

Reply:
Originally there were more tests put in that test stuff such as command validation (e.g `TestCommandSuccessful` or `TestCommandSuccessWithWhitespaces`).

After I refactored to follow the previous comments,  I was left just with the handler function. I will add more tests in to test the scenario in which the `nats.Request` receives an error (there are two types of errors, one from nats client and one from the response from slack). Originally there was no error checking in the `Request/Publish` step.

For the command, if it's not valid there is no behaviour, the handler function simply exists. Could perhaps add a log or something similar. @pamelin  what do you think?